src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -184,17 +190,48 @@ export const validateSubmission = async (
         req.formsg.featureFlags.includes(featureFlags.encryptionBoundaryShift),
       ),
     )
-    .map(() => next())
+    .map((parsedResponses) => {
+      const responses = [] as EncryptFormFieldResponse[]
+      for (const response of parsedResponses.getAllResponses()) {
+        if (response.isVisible) {
```

Comment:
qn: why is this check needed here and not in `email-submission.controller` (L181)? How does the email-submission controller deal with hidden fields

Reply:
Yea I suspect this is part of why we wanted to migrate to using `IncomingEncryptSubmission`, so that encapsulate all mutations required in the class itself, as described in the description of the PR https://github.com/opengovsg/FormSG/pull/1987. Not sure if we still want to continue this journey started 2 years back though? It does seem like a worthy investment if something similar to this encryption boundary shift is possible again 🤔 If not, I think we might have bigger fish to fry haha